using/self_service.md

```diff
@@ -322,3 +322,118 @@ Once allocation has been created, you can see its access information:
+
+Now you can proceed with Python management form and afterwards cotinue with [JupyterHub management](#jupyterhub-management-service).
+
+### Details screen elements
+
+![Python management details screen elements](../images/python-management-details.png)
+
+> Virtual environments are managed using virtualenv and virtualenvwrapper utilities.
+
+Python Management service is not notified when you install additional libraries manually.
+
+> If you install any libraries directly or create new virtual environments via SSH, click either Find missing libraries or Find installed virtual environments & libraries.
+
+1. Button to propagate changes to a virtual machine (install/delete libraries, create/delete virtual environments)
+2. Virtual environment name (a folder inside Virtual environments root directory)
+3. Download installed libraries list in a form of [requirements file format](https://pip.pypa.io/en/stable/user_guide/#requirements-files). Currently supports only exact versions of packages: pkg==1.1
+4. Upload requirements.txt for installation in a virtual environment. Again, only exact versions of packages. New libraries will appear in the form, save to apply the changes.
+5. If any libraries were installed manually using pip through SSH (or via Jupyter notebooks), you need to synchronize that virtual environment with ETAIS using this button (otherwise new libraries will not appear in a virtual environmnet)
+6. If any virtual environments were manually installed/removed in Virtual environments root directory, you need to synchronize the state with ETAIS with this button (otherwise new virtual environment will not appear).
+7. A brief guide which tells you how to activate virtual environment in terminal, install libraries and that you should click “Find missing installed libraries” button if new libraries were installed manually.
+8. Each action performed by ETAIS on a VM is logged here.
+9. Expand virtual environment (shows list of libraries)
+
+## JupyterHub management service
+
+Before using JupyterHub, please setup [Python management service](#python-management-service) on a VM.
+
+Once provisioned, JupyterHub configuration file is located in ```/etc/jupyterhub/jupyterhub_config.py```. All changes manually applied to JupyterHub configuraiton files will not be reflected in JupyterHub details form. Moreover, each time you click "Apply configuration" button, the existing configuration file is completely replaced with the new one generated out of the currently saved state in ETAIS. However, the backup for old configuration file is created in the same folder.
+
+JupyterHub management service is not notified if you change, for example, admin rights through JupyterHub GUI - do it through self-service.
+
+### Details screen - manual JupyterHub user entry
+
+![JupyterHub management details Linux PAM authentication section](../images/jupyterhub-management-linux-pam.png)
+
+1. Virtual environments come from Python management service.
```

Comment:
Why is enumeration needed here? The sentences to do not seem to make any sensible list.